game/blackearth.py

```diff
@@ -17,45 +23,59 @@ class BlackEarthGame(arcade.Window):
-            Vec2d(SCREEN_WIDTH/3,SCREEN_HEIGHT/3),
-            color=arcade.csscolor.DARK_RED
-            )
+    def setup(self, num_tanks=2):
+        """ Set up the game here. Call this function to restart the game.
         
-        self.player2 = Tank(
-            Vec2d(SCREEN_WIDTH*2/3, SCREEN_HEIGHT/3),
-            color=arcade.csscolor.DARK_BLUE
-        )
-        self.player2.turretAngleDeg = 90 + 45
+        Allow the user to specify how many tanks to start the game
+        with. Note that the way that we do this will DEFINITELY change
+        in the future, this is just a fun way to be able to parameterize
+
+
+        @param num_tanks Number of tanks to instantiate
+        """
+
+        # Set up players
+
+        # Populate a list based on number of tanks
+        # The list will be useful for keeping track of all of the tanks, and
+        # being able to do the same operation (such a "draw") on them all very
+        # easily.
+        self.tanksList = []
+        for n in range(1,num_tanks + 1):
+            new_tank = tank.Tank(
+                name=f"Player {n}",
+                position = pymunk.Vec2d(SCREEN_WIDTH*n/(num_tanks+1), SCREEN_HEIGHT/3),
+                color=next(tank.TANK_COLORS)
+            )
 
-    def on_key_press(self, key, modifiers):
-        # Pass through inputs to player1
-        self.player1.on_key_press(key, modifiers)
-    
-    def on_key_release(self, key, modifiers):
-        # Handle window events first
-        if key == arcade.key.ESCAPE:
-            exit()
+            self.tanksList.append(new_tank)
 
-        # Pass through inputs to player1
-        self.player1.on_key_release(key, modifiers)
+        # Create a circular Iterator for the tank list
+        # An Iterator is something different from a list, though certainly you can
+        # "iterate" through things like lists and tuples and even dictionaries.
+        # In other words, lists and tuples and dictionaries can themselves be termed
+        # "Iterators". I still want a list of tanks, but I also want something that
+        # points to that list, but I can use to cycle through them endlessly. I know
+        # that I want this, because I know that I just want to be able to keep going
+        # to whatever Tank's turn is next. Next, next, next. Me calling "next" shouldn't
+        # alter the list of tanks; I can have two "views" of the same data: the list
+        # view, and the endless cycle view.
+        self.tanksCycle = itertools.cycle(self.tanksList)
 
-    def on_update(self, delta_time):
-        # Update game state for game objects
-        self.player1.on_update()
+        # Set the active player
+        self.activeTank = next(self.tanksCycle)
+
+        # Set up other shapes to draw (just the ground for now)
         self.shapes = arcade.ShapeElementList()
-        
-        # Create the ground
+
+        # Create the ground (just a rectangle for now)
         ground = arcade.create_rectangle_filled(
```

Comment:
Hey @TheSpy0 ! Just wanted to point out that you can actually create comments _inside of a pull request attached to code_. Feel free to do so, either to make suggestions or just to ask questions. Example: "why the hell are you doing this right here?"

Also feel free to let me know if I'm overdoing this and bugging you. I'm just super excited to share all of the great things that other people have come up with to make lives easier. It's like smart people actually worked on this stuff!